src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -69,6 +69,20 @@ public async Task<SwaggerDocument> GenerateAsync(ApiDescriptionGroupCollection a
                 .GroupBy(item => item.Item2.ControllerTypeInfo.AsType())
                 .ToArray();
 
+            document.Consumes = apiGroups
+                .SelectMany(s => s.SupportedRequestFormats)
+                .Select(s => s.MediaType)
+                .Where(m => apiGroups.All(a => !a.SupportedRequestFormats.Any() || a.SupportedRequestFormats.Contains(m)))
```

Comment:
Why is it ok when empty (!any)? Is this correct?

Reply:
You're right. I think this condition is redundant